classes/04class/aws_iam_storage/README.md

```diff
@@ -0,0 +1,540 @@
+
+Information about which profiles are already configured can be obtained in the AWS CLI configuration file, usually in `~/.aws/config`, try:
+
+`cat ~/.aws/config`
+
+Credentials are in:
+
+`cat ~/.aws/credentials`
+
+You will see something like this, every word between brackets refers to an AWS profile followed by its credentials:
+```
+[default]
+aws_access_key_id = <MY_SECRET_ID>
+aws_secret_access_key = <MY_SECRET_KEY>
+
+[contino-denis]
+aws_access_key_id = <MY_SECRET_ID>
+aws_secret_access_key = MY_SECRET_KEY
+
+[denisdev]
+aws_access_key_id = <MY_SECRET_ID>
+aws_secret_access_key = QMY_SECRET_KEY
+```
+
+For using a specific profile during a terminal session, you can set the environment variable `AWS_PROFILE`:
+
+```
+export AWS_PROFILE=denisdev
+```
+
+#### Alternative method to set credentials (not recommended)
+
+You can also set the environment variables below to set up the credentials:
+
+```
+export AWS_ACCESS_KEY_ID=AKIAI44QH8DHBEXAMPLE
+export AWS_SECRET_ACCESS_KEY=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+export AWS_SESSION_TOKEN=AQoDYXdzEJr...<remainder of security token>
+```
+
+You can unset it with:
+```
+unset AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY AWS_SESSION_TOKEN AWS_SECURITY_TOKEN
+```
+
+#### Checking who are you with AWS Security Token Service (STS)
+```bash
+aws sts get-caller-identity
+```
+Output:
+```
+{
+    "UserId": "AIDAUDBAHX4WB4YEJPINL",
+    "Account": "281387974444",
+    "Arn": "arn:aws:iam::281387974444:user/denismac"
+}
+```
+
+## CLI Documentation and Help
+
+Usually any click that you perform in the AWS Console will have a correspondent AWS CLI command (be curious, use the `help` command a lot!).
+
+Try any command or subcommand followed by `help`:
+```
+aws help
+```
+or
+```
+aws s3 help
+```
+or
+```
+aws s3 cp help
+```
+
+You can also check the [web documentation.](https://docs.aws.amazon.com/cli/latest/index.html)
+
+## Example: List your S3 buckets
+
+> The credentials configured must be allowed to `s3:ListBucket` 
+
+> https://docs.aws.amazon.com/cli/latest/reference/s3api/list-buckets.html
+
+Run:
+
+`aws s3 ls`
+
+IAM will validate if your user has rights to perform this operation, if so AWS will return a list of buckets in the account.
+
+***
+
+# AWS Identity and Access Management (IAM)
+
+## What's IAM?
+AWS Identity and Access Management (IAM) is a web service that helps you securely control access to AWS resources. You use IAM to control who is **authenticated** (signed in) and **authorized** (has permissions) to use resources.
+
+Example: How do you grant read-only access to your friend's AWS user for your EC2 instances?
+
+## IAM - Four Pillars (Users, Groups, Roles, Policies)
+
+![](./assets/iam_overview.jpg)
+
+### Users
+An AWS Identity and Access Management (IAM) user is an entity that you create in AWS to represent the person or application that uses it to interact with AWS. A user in AWS consists of a name and credentials.
```

Comment:
You already created the link between Identity and Access Management to IAM above. No need to recreate id. You can just use `AWS IAM User` here